ci-operator/step-registry/telco5g/jobs-router/telco5g-jobs-router-commands.sh

```diff
@@ -10,45 +10,81 @@ echo "************ telco cluster setup command ************"
-echo "export DEBUG_JOB_TYPE=${JOB_TYPE}" >> $MAINENV
-echo "export DEBUG_RELEASE_IMAGE_LATEST=${RELEASE_IMAGE_LATEST-}" >> $MAINENV
-echo "export GIT_COMMITTER_NAME='CI User'" >> $MAINENV
-echo "export GIT_COMMITTER_EMAIL='cnf-devel@redhat.com'" >> $MAINENV
-echo "#######################################################" >> $MAINENV
+cat << EOF > "$MAINENV"
+# THERE IS THE ENVS FILE FOR JOBS
+export DEBUG_T5CI_JOB_TYPE='${T5CI_JOB_TYPE}'
+export DEBUG_T5CI_VERSION='${T5CI_VERSION}'
+export DEBUG_PROW_JOB_ID='${PROW_JOB_ID}'
+export DEBUG_JOB_NAME='${JOB_NAME}'
+export DEBUG_JOB_TYPE='${JOB_TYPE}'
+export DEBUG_RELEASE_IMAGE_LATEST='${RELEASE_IMAGE_LATEST-}'
+export GIT_COMMITTER_NAME='CI User'
+export GIT_COMMITTER_EMAIL='cnf-devel@redhat.com'
+#######################################################
+EOF
 
 if [[ "$PROW_JOB_ID" = *"nightly"* ]] && [[ "$JOB_TYPE" == "periodic" ]]; then
-    echo "export T5_JOB_TRIGGER=nightly" >> $MAINENV
-    echo "export T5_JOB_DESC=nightly-${T5CI_JOB_TYPE}" >> $MAINENV
-    echo "export T5_JOB_DESC_FULL=nightly-${T5CI_JOB_TYPE}-${T5CI_VERSION}" >> $MAINENV
     if [[ -n "${RELEASE_IMAGE_LATEST-}" ]]; then
         export IMG=${RELEASE_IMAGE_LATEST-}
     elif [[ -n "${RELEASE_IMAGE_INITIAL-}" ]]; then
         export IMG=${RELEASE_IMAGE_INITIAL-}
     else
         export IMG=${PROW_JOB_ID/-telco5g/}
     fi
-    echo "export T5_JOB_IMAGE=${IMG}" >> $MAINENV
-    # In case of running on nightly releases we need to figure out what release exactly to use
-    echo "export KCLI_PARAM=\"-P openshift_image=registry.ci.openshift.org/ocp/release:${IMG}\"" >> $MAINENV
-    echo "export SNO_PARAM=\"-f ${IMG}\"" >> $MAINENV
+    IMG_URL="registry.ci.openshift.org/ocp/release:$IMG"
+
+    cat << ____EOF >> "$MAINENV"
+export T5_JOB_TRIGGER=nightly
+export T5_JOB_DESC='nightly-${T5CI_JOB_TYPE}'
+export T5_JOB_DESC_FULL='nightly-${T5CI_JOB_TYPE}-${T5CI_VERSION}'
+# In case of running on nightly releases we need to figure out what release exactly to use
+export T5_JOB_RELEASE_IMAGE='$IMG_URL'
+export KCLI_PARAM='-P openshift_image=$IMG_URL'
+export SNO_PARAM='-f ${IMG}'
+____EOF
+
 elif [[ "$JOB_TYPE" == "periodic" ]]; then
-    echo "export T5_JOB_TRIGGER=periodic" >> $MAINENV
-    echo "export T5_JOB_DESC=periodic-${T5CI_JOB_TYPE}" >> $MAINENV
-    echo "export T5_JOB_DESC_FULL=periodic-${T5CI_JOB_TYPE}-${T5CI_VERSION}" >> $MAINENV
-    echo "export KCLI_PARAM=\"-P tag=${T5CI_VERSION} -P version=nightly\"" >> $MAINENV
-    echo "export SNO_PARAM=\"-t ${T5CI_VERSION} -r nightly\"" >> $MAINENV
+    # Identify release image url
+    # Ref.: https://github.com/karmab/kcli/blob/main/kvirt/cluster/openshift/disconnected/scripts/03_mirror.sh
+    if [ "$(echo "$T5CI_VERSION" | tr -dc '.' | wc -c)" -ge 2 ]; then
+        TAG="$T5CI_VERSION"
+    else
+        TAG="latest-$T5CI_VERSION"
+    fi
+    RELEASE_SUMMARY_URL="https://mirror.openshift.com/pub/openshift-v4/clients/ocp-dev-preview/$TAG/release.txt"
```

Comment:
`ocp-dev-preview` is a wrong link, we need to use `https://amd64.ocp.releases.ci.openshift.org/api/v1/releasestream/%s.0-0.%s/latest` like [here](https://gitlab.cee.redhat.com/telco-5g-devops-ci-infra/telco5g-lab-deployment/-/blob/master/scripts/sno_ag.py#L29) in this [function](https://gitlab.cee.redhat.com/telco-5g-devops-ci-infra/telco5g-lab-deployment/-/blob/master/scripts/sno_ag.py#L412-422)
So it will be like `https://amd64.ocp.releases.ci.openshift.org/api/v1/releasestream/4.13.0-0.nightly/latest` and JSON inside.

Reply:
The previous code passed `-P tag=${T5CI_VERSION} -P version=nightly` to `kcli`. Because version is `nightly`, the branch with code `OCP_REPO={{ 'ocp-dev-preview' if version == 'nightly' else 'ocp' }}` is being executed [here](https://github.com/karmab/kcli/blob/main/kvirt/cluster/openshift/disconnected/scripts/03_mirror.sh) (in case image mirroring is used). Could you point me to the code where `kcli` uses your urls?

Reply:
Maybe you got it from [here](https://github.com/karmab/kcli/blob/2acaffa19861be310e76f34aa09e6f4757d41566/kvirt/cluster/openshift/disconnected/scripts/03_mirror.sh#L7C46-L7C63)?

Reply:
Worth to ask @karmab about this. We fixed this logic once [here](https://github.com/karmab/kcli/blob/2acaffa19861be310e76f34aa09e6f4757d41566/kvirt/cluster/openshift/__init__.py#L397-L399). Since then we install a real nightly every job.

Reply:
seems like we don't use this script, but this one - [04_disconnected_mirror.sh](https://github.com/karmab/kcli-openshift4-baremetal/blob/main/scripts/04_disconnected_mirror.sh)
```
[root@cnfdu3-installer ~]# ls scripts/
00_virtual.sh              02_packages.sh  04_disconnected_mirror.sh  04_disconnected_quay.sh      05_nbde.sh  07_deploy_openshift.sh  09_post_install.sh  launch_steps.sh            notify.sh
01_patch_installconfig.sh  03_cache.sh     04_disconnected_olm.sh     04_disconnected_registry.sh  06_ntp.sh   08_nfs.sh               10_apps.sh          mirror-config.yaml.sample  oauth.yml
```

Reply:
Done.